src/js/CheckboxTree.js

```diff
@@ -337,7 +448,22 @@ class CheckboxTree extends React.Component {
             nodes,
             nativeCheckboxes,
         } = this.props;
-        const treeNodes = this.renderTreeNodes(nodes);
+
+        // reset for this render - values set in renderTreeNodes()
+        this.parents = {};
+
+        let root;
+        if (Array.isArray(nodes)) {
```

Comment:
I mentioned this elsewhere, but I do not really see the point of a root object. Is this used for some types of calculation? I would strongly prefer that such a root-level object need not exist.

Reply:
Covered in previous discussion...